File: arch/tv_ui_type/ui.java
package arch.tv_ui_type;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

class TVMenuItem {
    String name;
    TVMenuItem(String name) {
        this.name = name;
    }
}

public class UI {
    private List<TVMenuItem> menu = new ArrayList<>();
    private int selectedIndex = 0;

    public UI() {
        menu.add(new TVMenuItem("Live TV"));
        menu.add(new TVMenuItem("Apps"));
        menu.add(new TVMenuItem("Settings"));
        menu.add(new TVMenuItem("About CXXDROID"));
    }

    public void render() {
        System.out.println("\n=== CXXDROID TV Interface ===");
        for (int i = 0; i < menu.size(); i++) {
            if (i == selectedIndex)
                System.out.println(" > " + menu.get(i).name + " <");
            else
                System.out.println("   " + menu.get(i).name);
        }
        System.out.println("=============================");
    }

    public void navigate(String direction) {
        if (direction.equals("up") && selectedIndex > 0)
            selectedIndex--;
        else if (direction.equals("down") && selectedIndex < menu.size() - 1)
            selectedIndex++;
    }

    public void select() {
        System.out.println("Opening: " + menu.get(selectedIndex).name);
    }

    public static void main(String[] args) {
        UI ui = new UI();
        Scanner scanner = new Scanner(System.in);
        ui.render();
        System.out.println("Use commands: up / down / select / exit");

        while (true) {
            System.out.print("> ");
            String cmd = scanner.nextLine();
            if (cmd.equals("exit")) break;
            if (cmd.equals("up") || cmd.equals("down"))
                ui.navigate(cmd);
            else if (cmd.equals("select"))
                ui.select();
            ui.render();
        }

        System.out.println("Exiting TV UI.");
        scanner.close();
    }
}

File: arch/tv_ui_type/ui.cpp
#include <iostream>
#include <vector>
#include <string>

class TVMenuItem {
public:
    std::string name;
    TVMenuItem(const std::string& n) : name(n) {}
};

class TVUI {
private:
    std::vector<TVMenuItem> menu;
    int selectedIndex = 0;

public:
    TVUI() {
        menu.push_back(TVMenuItem("Live TV"));
        menu.push_back(TVMenuItem("Apps"));
        menu.push_back(TVMenuItem("Settings"));
        menu.push_back(TVMenuItem("About CXXDROID"));
    }

    void render() {
        std::cout << "\n=== CXXDROID TV Interface ===\n";
        for (int i = 0; i < menu.size(); ++i) {
            if (i == selectedIndex)
                std::cout << " > " << menu[i].name << " <\n";
            else
                std::cout << "   " << menu[i].name << "\n";
        }
        std::cout << "=============================\n";
    }

    void navigate(const std::string& direction) {
        if (direction == "up" && selectedIndex > 0)
            selectedIndex--;
        else if (direction == "down" && selectedIndex < (int)menu.size() - 1)
            selectedIndex++;
    }

    void select() {
        std::cout << "Opening: " << menu[selectedIndex].name << "\n";
    }
};

int main() {
    TVUI ui;
    std::string cmd;

    ui.render();
    std::cout << "Use commands: up / down / select / exit\n";

    while (true) {
        std::cout << "> ";
        std::cin >> cmd;
        if (cmd == "exit") break;
        if (cmd == "up" || cmd == "down") ui.navigate(cmd);
        else if (cmd == "select") ui.select();
        ui.render();
    }

    std::cout << "Exiting TV UI.\n";
    return 0;
}
